server/planning/events/utils.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+from eve.utils import ParsedRequest
+import json
+
+from planning.common import (
+    WORKFLOW_STATE,
+    get_max_recurrent_events,
+)
+from planning.events import EventsAsyncService
+from superdesk.resource_fields import ID_FIELD
+from superdesk.utc import utcnow
+
+
+async def get_series(query, sort, max_results):
+    events_service = EventsAsyncService()
+    page = 1
+
+    while True:
+        # Get the results from mongo
+        req = ParsedRequest()
+        req.sort = sort
+        req.where = json.dumps(query)
+        req.max_results = max_results
+        req.page = page
+        results = await events_service.get_from_mongo(req=req, lookup=None)
+
+        docs = list(results)
+        if not docs:
+            break
+
+        page += 1
+
+        # Yield the results for iteration by the callee
+        for doc in docs:
+            yield doc
+
+
+async def get_recurring_timeline(
```

Comment:
Would be good to have the signature of this function typed, including the return type (tuple of 3 dicts)